flink-ml-benchmark/src/test/java/org/apache/flink/ml/benchmark/BenchmarkTest.java

```diff
@@ -0,0 +1,185 @@
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.benchmark;
+
+import org.apache.flink.ml.benchmark.data.InputDataGenerator;
+import org.apache.flink.ml.benchmark.data.common.DenseVectorGenerator;
+import org.apache.flink.ml.clustering.kmeans.KMeans;
+import org.apache.flink.ml.clustering.kmeans.KMeansModel;
+import org.apache.flink.ml.param.WithParams;
+import org.apache.flink.ml.util.ReadWriteUtils;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.apache.flink.test.util.AbstractTestBase;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.apache.flink.ml.util.ReadWriteUtils.OBJECT_MAPPER;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Tests benchmarks. */
+@SuppressWarnings("unchecked")
+public class BenchmarkTest extends AbstractTestBase {
+    @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private static final String exampleConfigFile = "kmeansmodel-benchmark.json";
+    private static final String expectedBenchmarkName = "KMeansModel-1";
+
+    private final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+
+    @Before
+    public void before() {
+        System.setOut(new PrintStream(outputStream));
+    }
+
+    @After
+    public void after() {
+        System.setOut(originalOut);
+        outputStream.reset();
+    }
+
+    @Test
+    public void testCreateAndExecute() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+        KMeans kMeans = new KMeans().setK(5).setFeaturesCol("test_feature");
+        InputDataGenerator<?> inputsGenerator =
+                new DenseVectorGenerator()
+                        .setColNames("test_feature")
+                        .setNumValues(1000)
+                        .setVectorDim(10);
+
+        BenchmarkResult result =
+                BenchmarkUtils.runBenchmark(tEnv, "testBenchmarkName", kMeans, inputsGenerator);
+
+        BenchmarkUtils.printResults(result);
+        assertTrue(outputStream.toString().contains("testBenchmarkName"));
+        assertTrue(outputStream.toString().contains(result.totalTimeMs.toString()));
+    }
+
+    @Test
+    public void testLoadAndSave() throws Exception {
```

Comment:
I am hoping we can simply the test code here and make it more readable.

The following logical steps are executed by `flink-ml-benchmark.sh`:
- Parse command line argument
- Parse the json config file into `Map<String, ?>`
- Enumerate the `Map<String, ?>` and call `runBenchmark()` to get a list of BenchmarkResults.
- Output the BenchmarkResults to file or stdout.

Would it make the code more readable to update Benchmark.java and make its functions logically similar the the steps described above? For example, we can have functions such as `Map<String, ?> parseJsonFile(String path)`.

And in general the best practice to test `flink-ml-benchmark.sh` is to test each step separately by testing the corresponding function (e.g. `runBenchmark`, `parseJsonFile`).

Also, we might not need to explicitly test the step that converts BenchmarkResult to a string, as the corresponding function is quite straightforward and unlikely to have bug.